third_party/accessibility/ax/platform/ax_platform_node_mac.mm

```diff
@@ -704,18 +704,13 @@ - (NSString*)AXSelectedTextInternal {
 }
 
 - (NSValue*)AXSelectedTextRangeInternal {
-  // Selection might not be supported. Return (NSRange){0,0} in that case.
-  int start = 0, end = 0;
-  if (_node->IsPlainTextField()) {
-    start = _node->GetIntAttribute(ax::mojom::IntAttribute::kTextSelStart);
-    end = _node->GetIntAttribute(ax::mojom::IntAttribute::kTextSelEnd);
-  }
+  int start = _node->GetIntAttribute(ax::mojom::IntAttribute::kTextSelStart);
+  int end = _node->GetIntAttribute(ax::mojom::IntAttribute::kTextSelEnd);
   NSAssert((start >= 0 && end >= 0) || (start == -1 && end == -1), @"selection is invalid");
```

Comment:
This is a good example of the sort of asssertion I'd suggest adding at the spot I mentioned above.